code.c

```diff
@@ -0,0 +1,231 @@
+Array *create_array (int capacity) {
+  // Allocate memory for the Array struct
+  Array *newArr = malloc(sizeof(Array)); // allocates memory for a size of 1 struct Array
+  
+  // Set initial values for capacity and count
+  newArr->capacity = capacity;
+  newArr->count = 0;
+
+  // Allocate memory for elements
+  // hint: use `calloc`, code from lecture
+  newArr->elements = calloc(capacity, sizeof(char *));
+
+  return newArr;
+}
+
+
+/*****
+ * Free memory for an array and all of its stored elements
+ *****/
+void destroy_array(Array *arr) {
+  // Free all elements
+  free(arr->elements);
+
+  // Free array
+    free(arr); 
+}
+
+/*****
+ * Create a new elements array with double capacity and copy elements
+ * from old to new
+ *****/
+void resize_array(Array *arr) {
+  // Create a new element storage with double capacity
+  char **doubleCapacity = calloc((arr->capacity * 2), sizeof(char *));
+
+  // Copy elements into the new storage
+  for (int i=0; i< arr->count; i++) {
+    doubleCapacity[i] = arr->elements[i];
+  }
+
+  // Free the old elements array (but NOT the strings they point to)
+  if (arr->elements != NULL) {
+    free(arr->elements);
+  }
+
+  // Update the elements and capacity to new values
+  arr->elements = doubleCapacity; // set elements array with newly, expanded array
+  arr->capacity *= 2; // set the new capacity by 2
+}
+
+
+/************************************
+ *
+ *   ARRAY FUNCTIONS
+ *
+ ************************************/
+
+/*****
+ * Return the element in the array at the given index.
+ *
+ * Throw an error if the index is out of range.
+ *****/
+// fprintf writes formatted text to the output stream you specify.
+// printf is equivalent to writing fprintf(stdout, ...) and
+// writes formatted text to wherever the standard output stream is currently pointing.
+// sprintf writes formatted text to an array of char, as opposed to a stream.
+char *arr_read(Array *arr, int index) {
+  // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    fprintf(stderr, "Index is out of range.");
+    exit(1);
+  }
+
+  // Otherwise, return the element at the given index
+  return arr->elements[index];
+}
+
+
+/*****
+ * Insert an element to the array at the given index
+ *****/
+void arr_insert(Array *arr, char *element, int index) {
+  // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    fprintf(stderr, "Index is out of range.");
+    exit(1);
+  }
+
+  // Resize the array if the number of elements is equal to or over capacity
+  if (arr->count >= arr->capacity) {
+    resize_array(arr);
+  }
+
+  // Move every element after the insert index to the right one position
+  for (int i=arr->count; i > index; i--) {
+    arr->elements[i] = arr->elements[i-1]; // repointing the new address to previous element's address
```

Comment:
since you're decrementing towards the right, its a good thing you spotted it in your for loop, great work!